src/vfio/core.c

```diff
@@ -71,8 +83,47 @@ static int vfio_init_container(struct vfio_container *vfio)
 		return -1;
 	}
 
+	vfio->config_iommu = vfio_container_configure_iommu;
+	vfio->get_dev_fd = vfio_container_get_dev_fd;
+	vfio->do_dma_map = vfio_container_do_map_dma;
+	vfio->do_dma_unmap = vfio_container_do_unmap_dma;
+
 	return 0;
+}
+
+static int vfio_init_iommufd(struct vfio_container *vfio)
```

Comment:
In regards to my earlier point about refactoring. I think we would benefit from moving the iommufd implementation into its own file, and converge on a more generic interface to avoid having a function `vfio_init_vfio` and `vfio_init_iommufd`. The latter is hinting at being a vfio function, while the first becomes a little less descriptive (vs. `vfio_init_container`).

Reply:
I think these two will go away with the change that I'm planning of introducing a compile time ifdef that decides between iommufd and vfio backends.

Reply:
This is done. Will hit resolve